internal/authz/authz.go

```diff
@@ -195,6 +276,49 @@ func (a *Authorizer) individualAuthorize(ctx context.Context, clientID, relation
 	return nil
 }
 
+// moduleAuthorize checks if the user has access to each of the modules, and exits if an error is encountered.
+func (a *Authorizer) moduleAuthorize(ctx context.Context, clientID, relation, storeID string, modules []string) error {
+	var err error
+	var wg sync.WaitGroup
+	errorChannel := make(chan error, len(modules))
+	done := make(chan struct{})
+
+	for _, module := range modules {
+		wg.Add(1)
+		go func(module string) {
+			defer wg.Done()
+			contextualTuples := openfgav1.ContextualTupleKeys{
+				TupleKeys: []*openfgav1.TupleKey{
+					{
+						User:     StoreIDType(storeID).String(),
+						Relation: StoreType,
+						Object:   ModuleIDType(storeID).String(module),
+					},
+				},
+			}
+
+			err := a.individualAuthorize(ctx, clientID, relation, ModuleIDType(storeID).String(module), &contextualTuples)
+
+			if err != nil {
+				errorChannel <- err
+			}
+		}(module)
+	}
+
+	go func() {
+		wg.Wait()
+		close(done)
+		close(errorChannel)
+	}()
+
+	select {
+	case err = <-errorChannel:
+		return err
+	case <-done:
+		return nil
+	}
+}
```

Comment:
```suggestion
func (a *Authorizer) moduleAuthorize(ctx context.Context, clientID, relation, storeID string, modules []string) []error {
	var wg sync.WaitGroup
	errorChannel := make(chan error, len(modules))

	for _, module := range modules {
		wg.Add(1)
		go func(module string) {
			defer wg.Done()
			contextualTuples := openfgav1.ContextualTupleKeys{
				TupleKeys: []*openfgav1.TupleKey{
					{
						User:     StoreIDType(storeID).String(),
						Relation: StoreType,
						Object:   ModuleIDType(storeID).String(module),
					},
				},
			}

			err := a.individualAuthorize(ctx, clientID, relation, ModuleIDType(storeID).String(module), &contextualTuples)

			if err != nil {
				errorChannel <- err
			}
		}(module)
	}

	wg.Wait()
	close(errorChannel)

	errors := make([]error, len(modules))
	var i int

	for err := range errorChannel {
		if err != nil {
			errors[i] = err
			i++
		}
	}
	return errors[:i]
}
```

Reply:
@senojj : I think the original idea was to have early exit whenever there are any error.  With the suggested code, you will need to wait until all go routines are done.  in reality, I think what you propose is cleaner and at the bigger scope of things, the delta in latency should not be huge.  Just want to point it out the subtle difference here.

Reply:
ah ok, understood. In that case, a separate context would need to be passed into all of the go routines that could be canceled after the first routine completes. However, we would still need to wait for all routines to synchronize before returning from the function.

Reply:
For now I don't think we need to early exit since we don't really know if users will have many modules. I will apply this change, with the exception that right now we will just return the first error that is encountered, rather than all the errors as a slice